src/hs_validate.c

```diff
@@ -0,0 +1,35 @@
+#include "hs.h"
+#include <string.h>
+
+hs_error_t hs_validate(const char* pattern, unsigned int flags, char* err_msg, int err_msg_buffer_len) {
```

Comment:
I think `hs_validate` only checks the result of `hs_compile` which isn't quite qualified as a new API. It'll better if you implement this checking externally instead of inside Hyperscan.

Reply:
Done

Reply:
Thanks for your quick fix. The overall code changes look good to me.
We'll merge your commits into master as a single commit in next release coming soon. As a next step, we need to investigate and fix AVX2 issue together.